config/src/config/mod.rs

```diff
@@ -94,6 +94,17 @@ pub struct BaseConfig {
     data_dir: PathBuf,
     pub role: RoleType,
     pub waypoint: WaypointConfig,
+    genesis_waypoint: Option<WaypointConfig>,
+}
+
+impl BaseConfig {
+    pub fn genesis_waypoint(&self) -> Waypoint {
+        if self.genesis_waypoint.is_none() {
+            self.waypoint.waypoint()
+        } else {
+            self.genesis_waypoint.as_ref().unwrap().genesis_waypoint()
+        }
```

Comment:
better to do the following
```
if let Some(genesis_waypoint) = &self.genesis_waypoint {
  genesis_waypoint.waypoint()
} else {
  self.waypoint.waypoint()
}
```

Reply:
also this is still a breaking change, since historically we didn't have to define a genesis waypoint for FromStorage, so the right behavior is:

1. check if we have a waypoint defined for genesis_waypoint, if we do use it
2. if we don't and Waypoint is FromStorage use the current code `self.waypoint.genesis_waypoint()`

Reply:
@davidiw I am confused, this might be because I am new to rust or because there are different meanings of the same terms:
FromStorage is a WaypointConfig and it seems none of the WaypointConfig enum take genesis waypoint, so did you mean historically BaseConfig didn't have a genesis_waypoint field (which I added now)?

1. check if we have a waypoint defined for genesis_waypoint, if we do use it - I am assuming it means  check if BaseConfig has some value defined for genesis_waypoint 
2. if we don't and Waypoint is FromStorage use the current code self.waypoint.genesis_waypoint() - I am assuming even if Waypoint is not FromStorage, we need to call the current code self.waypoint.genesis_waypoint(),  because it FromStorage check in it

I am creating a commit with these interpretations. If it is still not right, it will really help if we can get on a call and resolve it tomorrow. I had sent a place holder meeting request yesterday.